libtransmission/peer-io.cc

```diff
@@ -267,10 +283,9 @@ static void event_write_cb(evutil_socket_t fd, short /*event*/, void* vio)
         return;
     }
 
-    EVUTIL_SET_SOCKET_ERROR(0);
-    auto const n_written = io->outbuf.toSocket(fd, howmuch); // -1 on err, 0 on EOF
-    auto const err = EVUTIL_SOCKET_ERROR();
-    auto const should_retry = n_written == -1 && (err == 0 || err == EAGAIN || err == EINTR || err == EINPROGRESS);
+    tr_error* error = nullptr;
+    auto const n_written = io->outbuf.toSocket(fd, howmuch, &error);
+    auto const should_retry = (error == nullptr) || canRetryFromError(error->code);
```

Comment:
should_retry should be the opposite condition:
- `error != nullptr && canRetryFromError(error->code)`